src/main/java/org/opensearch/geospatial/ip2geo/action/PutDatasourceRequest.java

```diff
@@ -0,0 +1,147 @@
+    public static final ObjectParser<PutDatasourceRequest, Void> PARSER;
+    static {
+        PARSER = new ObjectParser<>("put_datasource");
+        PARSER.declareString((request, val) -> request.setEndpoint(val), ENDPOINT_FIELD);
+        PARSER.declareLong((request, val) -> request.setUpdateIntervalInDays(TimeValue.timeValueDays(val)), UPDATE_INTERVAL_IN_DAYS_FIELD);
+    }
+
+    /**
+     * Default constructor
+     * @param datasourceName name of a datasource
+     */
+    public PutDatasourceRequest(final String datasourceName) {
+        this.datasourceName = datasourceName;
+    }
+
+    /**
+     * Constructor with stream input
+     * @param in the stream input
+     * @throws IOException IOException
+     */
+    public PutDatasourceRequest(final StreamInput in) throws IOException {
+        super(in);
+        this.datasourceName = in.readString();
+        this.endpoint = in.readString();
+        this.updateIntervalInDays = in.readTimeValue();
+    }
+
+    @Override
+    public void writeTo(final StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(datasourceName);
+        out.writeString(endpoint);
+        out.writeTimeValue(updateIntervalInDays);
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException errors = new ActionRequestValidationException();
+        validateEndpoint(errors);
+        validateUpdateInterval(errors);
+        return errors.validationErrors().isEmpty() ? null : errors;
+    }
+
+    private void validateEndpoint(final ActionRequestValidationException errors) {
+        try {
+            URL url = new URL(endpoint);
+            url.toURI(); // Validate URL complies with RFC-2396
+            validateManifestFile(url, errors);
+        } catch (MalformedURLException | URISyntaxException e) {
+            log.info("Invalid URL format is provided", e);
+            errors.addValidationError("Invalid URL format is provided");
+        }
+    }
+
+    private void validateManifestFile(final URL url, final ActionRequestValidationException errors) {
+        DatasourceManifest manifest;
+        try {
+            manifest = DatasourceManifest.Builder.build(url);
+        } catch (Exception e) {
+            log.info("Error occurred while reading a file from {}", url, e);
+            errors.addValidationError(String.format(Locale.ROOT, "Error occurred while reading a file from %s", url));
+            return;
+        }
+
+        try {
+            new URL(manifest.getUrl()).toURI(); // Validate URL complies with RFC-2396
+        } catch (MalformedURLException | URISyntaxException e) {
+            log.info("Invalid URL format is provided for url field in the manifest file", e);
+            errors.addValidationError("Invalid URL format is provided for url field in the manifest file");
+            return;
+        }
+
+        if (manifest.getValidForInDays() <= updateIntervalInDays.days()) {
```

Comment:
nit: shall we flip if condition to align with error message?

Reply:
Done